src/code/actions.js

```diff
@@ -566,15 +566,28 @@ function _keepOffspring(index, success, interactionType, shouldKeepSourceDrake)
   };
 }
 
-export function keepOffspring(index, success, maxDrakes, shouldKeepSourceDrake) {
+export function keepOffspring(index, keptDrakesIndices, maxDrakes, shouldKeepSourceDrake) {
   return (dispatch, getState) => {
-    const { interactionType } = getState();
+    const { interactionType, drakes } = getState();
+
+    let offspringOrg = new BioLogica.Organism(BioLogica.Species.Drake, drakes[index].alleleString, drakes[index].sex),
+        success = true;
+    // Don't keep any drakes with the same phenotypes
+    keptDrakesIndices.forEach(function(keptDrakeIndex) {
```

Comment:
This is fine, since it's a small operation, but `every` might be a better function to use here, because it won't keep spinning around after you've already set `success = false`.

```
success = keptDrakesIndices.every(function(keptDrakeIndex) {
...
return keptImage !== offspringOrg.getImageName();
});
```